core/src/Streamly/Internal/Data/Parser.hs

```diff
@@ -409,12 +412,17 @@ eof = D.toParserK D.eof
 -- >>> Stream.parse (Parser.satisfy (== 1)) $ Stream.fromList [1,0,1]
 -- 1
 --
+-- >>> toMaybe f x = if f x then Just x else Nothing
+-- >>> satisfy f = Parser.maybe (toMaybe f)
+--
 -- /Pre-release/
 --
 {-# INLINE satisfy #-}
 satisfy :: MonadCatch m => (a -> Bool) -> Parser m a a
 satisfy = D.toParserK . D.satisfy
 
+-- XXX Change this to "next".
```

Comment:
What?

Reply:
To `next` or to `one`?

Reply:
`Parser.fromFold one` would have `Parser m a (Maybe a)`
Should we have a combinator that converts failable folds to failure in parsers?
`fromFoldMaybe :: Fold m a (Maybe b) -> Parser m a b`?
Can you see any other use case for this?